### PR TITLE
chore: Fix some linter warnings

### PR DESCRIPTION
**Pull request by keawade**
#### Description
A couple minor refactors to fix linter warnings in modules.

#### Motivation and Context
There are a lot of linter warnings. It looks like most of them came in with the recent deprecation of `new_segment`.

I've cleaned up a lot of them but figure I should do these fixes in batches so that I lower the risk of merge conflicts. I'll probably push some more batches of changes to the PR if I fix more linter issues before the PR gets reviewed.

#### Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### How Has This Been Tested?
I've been manually smoke testing and running `cargo test` religiously.

- [ ] I have tested using **MacOS**
- [x] I have tested using **Linux**
- [ ] I have tested using **Windows**

#### Checklist:
- [x] I have updated the documentation accordingly.
- [x] I have updated the tests accordingly.

### CODE REVIEW

File: src/modules/memory_usage.rs [hunk 1]
```diff
@@ -69,20 +51,19 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
...
+        module.create_segment(
             "memory_usage",
-            &format!("{} | {}", display_mem, display_swap),
+            &SegmentConfig::new("").with_value(&format!("{} | {}", display_mem, display_swap)),
         );
     } else {
-        module.new_segment("memory_usage", &display_mem);
+        module.create_segment(
+            "memory_usage",
+            &SegmentConfig::new("").with_value(&display_mem),
```
Comment (brianlow) [inline src/modules/memory_usage.rs]: Could this be `&SegmentConfig::new(&display_mem)`?

File: src/configs/git_status.rs [hunk 2]
```diff
@@ -0,0 +1,46 @@
+use crate::config::{ModuleConfig, RootModuleConfig, SegmentConfig};
```
Comment (brianlow) [inline src/configs/git_status.rs]: Also doing this in #434

File: src/modules/memory_usage.rs (hunk 1 above)
Comment (keawade) [reply src/modules/memory_usage.rs]: That makes a lot more sense :laughing:. I'll give it a try, thanks!

File: src/configs/git_status.rs (hunk 2 above)
Comment (keawade) [reply src/configs/git_status.rs]: Okay, cool. I've reverted my changes to the `git_status` module.

**Comment by keawade** [comment]
Closing this PR since these fixes have already been implemented in #514 and #515.

**Pull request closed without merging**